frontend/simple/controller/backend.js

```diff
@@ -0,0 +1,102 @@
+/* global fetch */
+'use strict'
+
+import sbp from '../../../shared/sbp.js'
+import {sign, bufToB64, b64ToStr} from '../../../shared/functions.js'
+import {GIMessage} from '../../../shared/events.js'
+import {RESPONSE_TYPE} from '../../../shared/constants.js'
+import pubsub from './utils/pubsub.js'
+import {handleFetchResult} from './utils/misc.js'
+
+// temporary identity for signing
+const nacl = require('tweetnacl')
+var persona = nacl.sign.keyPair()
+var signature = signJSON('', persona)
+
+function signJSON (json, keypair) {
+  return sign({
+    publicKey: bufToB64(keypair.publicKey),
+    secretKey: bufToB64(keypair.secretKey)
+  }, json)
+}
+
+var subscriptions = []
+var primus
+
+export function createWebSocket (url, options) {
+  return new Promise((resolve, reject) => {
+    primus = pubsub({
+      url,
+      options,
+      handlers: {
+        open: () => {
+          console.log('websocket connection opened!')
+          resolve(primus)
+        },
+        error: err => {
+          console.log('websocket error:', err.message, err)
+          reject(err)
+        },
+        data: msg => {
+          console.log('websocket message:', msg)
+          if (!msg.data) throw new Error('malformed message: ' + JSON.stringify(msg))
+          switch (msg.type) {
+            case RESPONSE_TYPE.ENTRY:
+              // calling dispatch via SBP makes it simple to implement 'test/backend.js'
+              sbp('state/vuex/dispatch', 'handleEvent', GIMessage.deserialize(msg.data))
+              break
+            default:
+              console.log('SOCKET UNHANDLED EVENT!', msg) // TODO: this
+          }
+        }
+      }
+      // TODO: handle going offline event
+    })
+    primus.on('reconnected', () => {
+      console.log('websocket connection re-established. re-joining:', subscriptions)
+      subscriptions.forEach(contractID => primus.sub(contractID))
+    })
+  })
+}
+
+// Keep pubsub in sync (logged into the right "rooms") with store.state.contracts
+sbp('okTurtles.events/on', 'contractsModified', async (data) => {
+  var contractID = data.add || data.remove
+  var idx = subscriptions.indexOf(contractID)
+  var method = data.add ? 'sub' : 'unsub'
+  if ((data.add && idx > -1) || (data.remove && idx === -1)) {
+    return // if already subscribed or already unsubscribed
```

Comment:
Usually a function should always return something, even if `false` only, that I believe it's what's intended here. `return false`

Reply:
This function doesn't return anything, so returning `undefined` is perfectly fine